lib/Doctrine/ODM/MongoDB/Persisters/DocumentPersister.php

```
@@ -1021,9 +1021,24 @@ public function prepareQueryOrNewObj(array $query, bool $isNewObj = false) : arr
 
             $preparedQueryElements = $this->prepareQueryElement((string) $key, $value, null, true, $isNewObj);
             foreach ($preparedQueryElements as [$preparedKey, $preparedValue]) {
-                $preparedQuery[$preparedKey] = is_array($preparedValue)
-                    ? array_map('\Doctrine\ODM\MongoDB\Types\Type::convertPHPToDatabaseValue', $preparedValue)
-                    : Type::convertPHPToDatabaseValue($preparedValue);
+                if (is_array($preparedValue)) {
+                    $preparedQuery[$preparedKey] = array_map('\Doctrine\ODM\MongoDB\Types\Type::convertPHPToDatabaseValue', $preparedValue);
+
+                    continue;
+                }
+
+                $preparedValue = Type::convertPHPToDatabaseValue($preparedValue);
+
+                if ($this->class->hasField($key)) {
+                    $mapping  = $this->class->fieldMappings[$key];
+                    $typeName = $mapping['type'];
+                    if (Type::hasType($typeName) && ! in_array($typeName, ['collection', 'hash'])) {
```

I would suggest throwing an exception if we can't find the type: this should've been caught during the metadata mapping phase and not be silently ignored here.

---

Done